.gitlab-ci.yml

```diff
@@ -468,6 +468,31 @@ build-macos-subkey:
   tags:
     - osx
 
+build-rust-doc:
+  stage:                           build
+  <<:                              *docker-env
+  <<:                              *test-refs
+  needs:
+    - job:                         test-linux-stable
+      artifacts:                   false
+  variables:
+    <<:                            *default-vars
+    SKIP_WASM_BUILD:               1
+    RUSTDOCFLAGS:                  "--html-in-header $(pwd)/.maintain/rustdoc-header.html -Dwarnings"
```

Comment:
as long as we have `-Dwarnings` doc will not build successfully because sc-network doc have some warning. (or I missed a PR).